codex-rs/login/src/lib.rs

```diff
@@ -261,11 +262,25 @@ pub struct SpawnedLogin {
     pub stderr: Arc<Mutex<Vec<u8>>>,
 }
 
+fn ensure_login_script(codex_home: &Path) -> std::io::Result<PathBuf> {
```

Comment:
Instead of writing to `CODEX_HOME`, can we use `tempfile::NamedTempFile` in the OS temp directory? The main thing I care about is that as part of `Drop`, it should be deleted.

I suppose that still works if the `tempfile::NamedTempFile` is created in `CODEX_HOME`, but it's possible it won't get cleaned up in the event of a panic, so I think we should stick with the temp folder unless there is a compelling reason not to.

Reply:
No you need to return the `NamedTempFile` from this function so it stays alive.